lib/tx.py

```diff
@@ -217,3 +217,11 @@ def read_tx(self):
 
         return TxSegWit(version, marker, flag, inputs,
                         outputs, witness, locktime), double_sha256(orig_ser)
+
+
+class DeserializerFairCoin(Deserializer):
+    def read_block(self):
+        '''Returns a list of (deserialized_tx, tx_hash) pairs.'''
+        read_tx = self.read_tx
+        txs = [read_tx() for n in range(self._read_varint())]
+        return txs
```

Comment:
Yes, it crashed because of the assertion.
A Bitcion block has this structure:
```
header (80 bytes)
list of transactions
```
A FairCoin block has this format:
```
header (108 bytes)
list of transactions
additional proof-of-cooperation payload
```

And this additional payload triggers the assertion. That's why I was forced to create a Deserializer.